core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -2073,8 +2118,10 @@ func (o *evmTxStore) FindConfirmedTxesReceipts(ctx context.Context, finalizedBlo
 	var cancel context.CancelFunc
 	ctx, cancel = o.stopCh.Ctx(ctx)
 	defer cancel()
+
+	// note the receipts are partially loaded for performance reason
 	err = o.Transact(ctx, true, func(orm *evmTxStore) error {
```

Comment:
This is slightly out of scope, but could we change Transact to Execute instead? I don't see a reason for a transaction here, we only have a single query.